### ISSUE TITLE
[Accessibility]: Tree page (post-redesign): Navigation links are reported as buttons

### ISSUE THREAD

**Issue by umpox** [sourcegraph/sourcegraph#35295]
### Audit type

Screen reader navigation

### User journey audit issue

https://github.com/sourcegraph/sourcegraph/issues/34422

### Problem description

<img width="868" alt="image" src="https://user-images.githubusercontent.com/9516420/167853787-1758ebeb-3ca6-412a-90fd-e8a2923cafef.png">

In the redesign these items should function as links (they navigate) and a navigation list This is not the case:
- Screen reader reports these as buttons
- Screen reader loses focus when a button is clicked

### Expected behavior

1. Remove role="button" from these links.
2. Wrap this links in `<ul>` and `<li>`
3. Wrap the list in a `<nav>` element - so it is reported as a navigation list to a screen reader
4. Check if focus is still lost when selecting a button using a keyboard:
  a. If still not fixed, we can manually move the focus to the new page through JS.

### Assigning labels

- [X] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [X] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [ ] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [ ] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

This issue will be fixed by my team, I have assigned a relevant member to it, or I will do so in the near future.

**Comment by umpox** [comment]
@gitstart-sourcegraph Please ensure the `new-repo-page` is feature flag is **enabled** for you before trying to make these changes. You should be able to check at : https://sourcegraph.test:3443/site-admin/surveys
